### PR TITLE
chore(downloader): replace workflow with a single request

### PR DESCRIPTION
**Pull request by cnishina**
- add method to downloader to getFile
- add method to file manager to use new downloader getFile
- rewrite updateBinary function to use new file manager method
- add a couple of unit tests around downloader getFile
- use es6 promises, change travis to use node 6 minimum
- remove old method to request file in file manager and downloader

### CODE REVIEW

**Comment by cnishina** [comment]
Note: needs some code clean up and notes to self for cleaning up (for another PR)

**Comment by cnishina** [comment]
Should merge https://github.com/angular/webdriver-manager/pull/160 before this and rebase this.

File: lib/cmds/update.ts [hunk 1]
```diff
@@ -66,7 +66,8 @@ if (argv._[0] === 'update-run') {
  * Parses the options and downloads binaries if they do not exist.
  * @param options
  */
-function update(options: Options): void {
+function update(options: Options) {
+  let promises: Promise<boolean>[] = [];
```
Comment (sjelin) [inline lib/cmds/update.ts]: Nothing is ever pushed to this array.  Is this still a WIP?

**Comment by sjelin** [comment]
Looks good once you rebase

**Pull request merged**